pandas/core/construction.py

```diff
@@ -565,3 +565,64 @@ def _try_cast(
         else:
             subarr = np.array(arr, dtype=object, copy=copy)
     return subarr
+
+
+# see gh-17261
+def is_empty_data(data):
```

Comment:
can you type this (returns bool), data could be Any

move the issue reference inside the function